[DOCS-8931] Strategies for Reducing Log Volume guide (#25126)
DataDog/documentation: content/en/observability_pipelines/guide/strategies_for_reducing_log_volume.md

---
title: Strategies for Reducing Log Volume
disable_toc: false
further_reading:
- link: "/observability_pipelines/set_up_pipelines"
  tag: "documentation"
  text: "Set up a pipeline"
- link: "/observability_pipelines/processors/"
  tag: "documemtation"
  text: "Observability Pipelines processors"
---

## Overview

Observability Pipelines allows you to collect and process logs, as well as determine where you want to route them, all before the data leaves your on-premises or cloud environment.

> The value of your logs is not all equal. For example, error logs are often more useful than info logs when troubleshooting live operations. Logs from production environments are also more important than logs from non-production environments. Therefore, routing all of your logs to an indexed solution can end up diluting the overall value of your data and cause you to exceed your budget.

Reviewer: >The value of your logs is not all equal.

This just reads funny to me, I am not an expert on grammar or wordsmithing, so grain of salt, but putting the `all` before `logs` reads better to me, i.e.

> The value of all your logs is not equal

? 

Not sure if that came from my doc, if so then just me talking to myself 😆

Author: Thanks for point that out! I was struggling with it before....it read funny to me too. I just updated it to: `Logs are not all equal in value`. What do you think?

> The value of your logs is not all equal. For example, error logs are often more useful than info logs when troubleshooting live operations. Logs from production environments are also more important than logs from non-production environments. Therefore, routing all of your logs to an indexed solution can end up diluting the overall value of your data and cause you to exceed your budget.

Author: ```suggestion
Not all logs have equal value. For example, error logs are often more useful than info logs when troubleshooting live operations. Logs from production environments are also more important than logs from non-production environments. Therefore, routing all of your logs to an indexed solution can end up diluting the overall value of your data and cause you to exceed your budget.
```

The following situations can also unnecessarily increase your log volume and cost:

> - An application team inadvertently turns on debug logging and a new build triggers a flood of errors.

Reviewer: These are two distinct scenarios, 1 is turning on debug logging, 2 is shipping a new build with a new error loop condition, WDYT about separating them? Or we focus on one scenario?

Author: Ah okay, I've split it up into two scenarios.

> - An application team inadvertently turns on debug logging and a new build triggers a flood of errors.

Author: ```suggestion
- An application team inadvertently turns on debug logging.
- A flood of errors is triggered when shipping a new build with an error loop condition.
```

> - Teams attempt to add as much performance and metric data into the logs because it's easier than separating out the data.

Reviewer: > easier than separating out the data

Maybe "easier than implementing other telemetry solutions"? (e.g. APM and metrics)

Author: updated!

> - Teams attempt to add as much performance and metric data into the logs because it's easier than separating out the data.

Author: ```suggestion
- Teams attempt to add as much performance and metric data into the logs because it seems easier than implementing other telemetry solutions.
```

### Filter your logs

Not all logs are valuable and need to be stored. For example, keeping debug logs from non-production systems is most likely not critical for your organization. Therefore, use the [filter processor][2] to drop those logs, so they do not get sent to your log management solutions.

### Drop attributes in your logs

Sometimes logs can contain hundreds of attributes, and often only a small number of attributes are used for investigation and analysis. Use the [edit fields][3] processor to reduce the overall size of your logs by dropping unused or non-useful attributes, thus lowering your log ingestion cost.

### Reduce your logs

Some systems can emit hundreds, if not thousands, of logs per second. Many of these logs can be collapsed into a single event by merging fields using different strategies, such as concatenation, summing, creating an array of the values, and more. Use the [reduce processor][4] to collapse multiple log events into one event, based on the selected merge strategy. This reduces the total number of events that get sent to your destination.

### Deduplicate your logs

> Deduplicating your logs can help maintain the accuracy and consistency of your data and protect against upstream mistakes that accidentally duplicates your logs. Use the [deduplication processor][5] to compare fields and see if there is identical content and then drop the duplicates, thus reducing your total log volume.

Author: The dedupe links is broken because I'm missing the dedupe processor reference doc. Opening a separate PR to add that.

Author: PR: https://github.com/DataDog/documentation/pull/25144

---
title: Strategies for Reducing Log Volume
disable_toc: false
further_reading:
- link: "/observability_pipelines/set_up_pipelines"
  tag: "documentation"
  text: "Set up a pipeline"
- link: "/observability_pipelines/processors/"
  tag: "documemtation"
  text: "Observability Pipelines processors"
---

## Overview

Observability Pipelines allows you to collect and process logs, as well as determine where you want to route them, all before the data leaves your on-premises or cloud environment.

> Not all logs have equal value. For example, error logs are often more useful than info logs when troubleshooting live operations. Logs from production environments are also more important than logs from non-production environments. Therefore, routing all of your logs to an indexed solution can end up diluting the overall value of your data and cause you to exceed your budget.

Author: ```suggestion
Logs are not equal value. For example, error logs are often more useful than info logs when troubleshooting live operations. Logs from production environments are also more important than logs from non-production environments. Therefore, routing all of your logs to an indexed solution can end up diluting the overall value of your data and cause you to exceed your budget.
```

## Strategies for reducing your log volume

Follow these strategies to reduce your log volume:

### Sample your logs

Reduce your overall log volume by dropping a percentage of high-volume logs that you know are repetitive, noisy, or less valuable. Use the [sample processor][1] to match a subset of logs based on the filter query and keep only the percentage of logs you've specified. This provides you a representative view of high volume log streams, while maintaining visibility and the ability for analysis.

### Filter your logs

Not all logs are valuable and need to be stored. For example, keeping debug logs from non-production systems is most likely not critical for your organization. Therefore, use the [filter processor][2] to drop those logs, so they do not get sent to your log management solutions.

### Drop attributes in your logs

Sometimes logs can contain hundreds of attributes, and often only a small number of attributes are used for investigation and analysis. Use the [edit fields][3] processor to reduce the overall size of your logs by dropping unused or non-useful attributes, thus lowering your log ingestion cost.

### Reduce your logs

> Some systems can emit hundreds, if not thousands, of logs per second. Many of these logs can be collapsed into a single event by merging fields using different strategies, such as concatenation, summing, creating an array of the values, and more. Use the [reduce processor][4] to collapse multiple log events into one event, based on the selected merge strategy. This reduces the total number of events that get sent to your destination.

Reviewer 2: ```suggestion
Systems can emit hundreds, if not thousands, of logs per second. Collapse these logs can be into a single event by merging fields using different strategies, such as concatenation, summing, creating an array of the values, and more. Use the [reduce processor][4] to collapse multiple log events into one event, based on the selected merge strategy. This reduces the total number of events that get sent to your destination.
```
